client/mainmenu/CPrologEpilogVideo.cpp

```diff
@@ -48,15 +57,21 @@ void CPrologEpilogVideo::show(Canvas & to)
 	//some videos are 800x600 in size while some are 800x400
 	CCS->videoh->update(pos.x, pos.y + (CCS->videoh->size().y == 400 ? 100 : 0), to.getInternalSurface(), true, false);
 
-	//move text every 5 calls/frames; seems to be good enough
-	++positionCounter;
-	if(positionCounter % 5 == 0)
+	const uint32_t speed = (voiceDurationMilliseconds == 0) ? 100 : (voiceDurationMilliseconds / (text->textSize.y));
+
+	if(elapsedTimeMilliseconds > speed && text->textSize.y - 50 > positionCounter)
+	{
 		text->scrollTextBy(1);
+		elapsedTimeMilliseconds -= speed;
+		++positionCounter;
+	}
 	else
+	{
 		text->showAll(to); // blit text over video, if needed
 
-	if(text->textSize.y + 100 < positionCounter / 5 && voiceStopped)
-		clickPressed(GH.getCursorPosition());
+		if(elapsedTimeMilliseconds > (voiceDurationMilliseconds == 0.0 ? 6000 : 3000) && voiceStopped) // pause after completed scrolling (longer for intros missing voice)
+			clickPressed(GH.getCursorPosition());
+	}
```

Comment:
Should've clarified this before.
`show` method should only well, show. Actual logic (like most of this block) should go to `tick` method